1.segment_images/scripts/0.segment_nuclei_organoids.py

```diff
@@ -0,0 +1,191 @@
+# coding: utf-8
+
+# This notebook focuses on trying to find a way to segment cells within organoids properly.
+# The end goals is to segment cell and extract morphology features from cellprofiler.
+# These masks must be imported into cellprofiler to extract features.
+
+# ## import libraries
+
+# In[1]:
+
+
+import argparse
+import pathlib
+
+import matplotlib.pyplot as plt
+
+# Import dependencies
+import numpy as np
+import skimage
+import tifffile
+from cellpose import core, models
+from skimage import io
+
+# check if in a jupyter notebook
+try:
+    cfg = get_ipython().config
+    in_notebook = True
+except NameError:
+    in_notebook = False
+
+
+# ## parse args and set paths
+
+# In[2]:
+
+
+if not in_notebook:
+    print("Running as script")
+    # set up arg parser
+    parser = argparse.ArgumentParser(description="Segment the nuclei of a tiff image")
+
+    parser.add_argument(
+        "--input_dir",
+        type=str,
+        help="Path to the input directory containing the tiff images",
+    )
+    parser.add_argument(
+        "--window_size", type=int, help="Size of the window to use for the segmentation"
+    )
+    parser.add_argument(
+        "--clip_limit",
+        type=float,
+        help="Clip limit for the adaptive histogram equalization",
+    )
+
+    args = parser.parse_args()
+    window_size = args.window_size
+    clip_limit = args.clip_limit
+    input_dir = pathlib.Path(args.input_dir).resolve(strict=True)
+else:
+    print("Running in a notebook")
+    input_dir = pathlib.Path("../examples/raw_z_input/").resolve(strict=True)
+    window_size = 3
+    clip_limit = 0.05
+
+mask_path = pathlib.Path(f"../processed_data/{input_dir.stem}").resolve()
+mask_path.mkdir(exist_ok=True, parents=True)
+
+
+# ## Set up images, paths and functions
+
+# In[3]:
+
+
+image_extensions = {".tif", ".tiff"}
+files = sorted(input_dir.glob("*"))
+files = [str(x) for x in files if x.suffix in image_extensions]
+
+
+# In[4]:
+
+
+for f in files:
+    if "405" in f:
+        nuclei = io.imread(f)
+nuclei = np.array(nuclei)
+imgs = skimage.exposure.equalize_adapthist(nuclei, clip_limit=clip_limit)
+original_imgs = imgs
+print("Subsampled image shape:", imgs.shape)
+print("number of z-slices:", imgs.shape[0])
+original_z_slice_count = len(imgs)
+print("number of z slices in the original image:", original_z_slice_count)
+
+
+# In[5]:
+
+
+# make a 2.5 D max projection image stack with a sliding window of 3 slices
+
+image_stack_2_5D = np.empty((0, imgs.shape[1], imgs.shape[2]), dtype=imgs.dtype)
+for image_index in range(imgs.shape[0]):
+    image_stack_window = imgs[image_index : image_index + window_size]
+    if not image_stack_window.shape[0] == window_size:
+        break
+    # max project the image stack
+    image_stack_2_5D = np.append(
+        image_stack_2_5D, np.max(image_stack_window, axis=0)[np.newaxis, :, :], axis=0
+    )
+
+image_stack_2_5D = np.array(image_stack_2_5D)
+imgs = np.array(image_stack_2_5D)
```

Comment:
Aren't both of these lines of code doing the same thing? I am a bit confused why you need to np.array the same variable twice. Here is my suggested change, but if there is a reason for this, recommend adding code comment:
```suggestion
imgs = np.array(image_stack_2_5D)
```